syz-verifier/stats/stats.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+// Package stats contains utilities that aid gathering statistics of
+// system call mismatches.
+package stats
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"sort"
+
+	"github.com/google/syzkaller/prog"
+)
+
+// Stats encapsulates data for creating statistics related to the results
+// of the verification process.
+type Stats struct {
+	// Calls stores the number of mismatches for each system call that can
+	// be contained by a program
+	Calls map[string]int
```

Comment:
How are you going to compute the % mismatches per syscall, if you only collect the mismatches?

Where is the total number of calls per syscall?


I think what we want is primarily stats on mismatching syscalls. Because you could have a very very large program, but only 1 syscall mismatched:
- let's assume 2 very large programs of 1000 syscalls each.
- let's say for each of them 1 syscall fails.

In your current version, what I see is that we'd end up with 100% mismatches, which is a very skewed view of reality. The reality would actually be only 0.1% mismatches.

You can still collect stats on coarser level of runs/programs as well.